packages/flutter_tools/lib/src/build_system/hash.dart

```diff
@@ -0,0 +1,171 @@
+  20, 05, 09, 14, 20, 05, 09, 14, 20, 05, 09, 14, 20, 04, 11, 16, 23, 04, 11,
+  16, 23, 04, 11, 16, 23, 04, 11, 16, 23, 06, 10, 15, 21, 06, 10, 15, 21, 06,
+  10, 15, 21, 06, 10, 15, 21
+];
+
+/// A bitmask that limits an integer to 32 bits.
+const int _mask32 = 0xFFFFFFFF;
+
+/// An incremental hash computation that approximates md5.
+///
+/// The byte finalization required for a true md5 is not implemented so the
+/// computations will not match up with other md5 packages.
+class Md5Hash {
+  Md5Hash() {
+    _digest[0] = 0x67452301;
+    _digest[1] = 0xefcdab89;
+    _digest[2] = 0x98badcfe;
+    _digest[3] = 0x10325476;
+  }
+
+  /// The current hash digest.
+  final Uint32List _digest = Uint32List(4);
+  final Uint8List _scratchSpace = Uint8List(512);
+  static const int _signatureBytes = 8;
+  int _remainingLength;
+  int _contentLength = 0;
+
+  void addChunk(Uint8List data, [int stop]) {
+    assert(_remainingLength == null);
+    stop ??= data.length;
+    int i = 0;
+    for (; i <= stop - 512; i += 512) {
+      final Uint32List view = Uint32List.view(data.buffer, i, 16);
+      _writeChunk(view);
+    }
+    if (i != stop) {
+      // The data must be copied so that the provided buffer can be reused.
+      int j = 0;
+      for (; i < stop; i += 1) {
+        _scratchSpace[j] = data[i];
+        j += 1;
+      }
+      _remainingLength = j;
+    }
+    _contentLength += stop;
+  }
+
+  void _writeChunk(Uint32List chunk) {
+    int a = _digest[0];
+    int b = _digest[1];
+    int c = _digest[2];
+    int d = _digest[3];
+    int e = 0;
+    int f = 0;
+    int i = 0;
+    for (; i < 16; i += 1) {
+      e = (b & c) | ((~b & _mask32) & d);
+      f = i;
+      final int temp = d;
+      d = c;
+      c = b;
+      b = _add32(
+          b,
+          _rotl32(_add32(_add32(a, e), _add32(_noise[i], chunk[f])),
+              _shiftAmounts[i]));
+      a = temp;
+    }
+    for (; i < 32; i += 1) {
+      e = (d & b) | ((~d & _mask32) & c);
+      f = ((5 * i) + 1) % 16;
+      final int temp = d;
+      d = c;
+      c = b;
+      b = _add32(
+          b,
+          _rotl32(_add32(_add32(a, e), _add32(_noise[i], chunk[f])),
+              _shiftAmounts[i]));
+      a = temp;
+    }
+    for (; i < 48; i += 1) {
+      e = b ^ c ^ d;
+      f = ((3 * i) + 5) % 16;
+      final int temp = d;
+      d = c;
+      c = b;
+      b = _add32(
+          b,
+          _rotl32(_add32(_add32(a, e), _add32(_noise[i], chunk[f])),
+              _shiftAmounts[i]));
+      a = temp;
+    }
+    for (; i < 64; i+= 1) {
+      e = c ^ (b | (~d & _mask32));
+      f = (7 * i) % 16;
+      final int temp = d;
+      d = c;
+      c = b;
+      b = _add32(
+          b,
+          _rotl32(_add32(_add32(a, e), _add32(_noise[i], chunk[f])),
+              _shiftAmounts[i]));
+      a = temp;
+    }
+
+    _digest[0] += a;
+    _digest[1] += b;
+    _digest[2] += c;
+    _digest[3] += d;
+  }
+
+  Uint32List finalize() {
```

Comment:
I did not implement the finalization where the byte length is also written into the signature